test/performance/benchmarks/load-test/main.go

```diff
@@ -0,0 +1,187 @@
+	defer cancel()
+
+	influxReporter, err := performance.NewInfluxReporter(map[string]string{"flavor": *flavor})
+	if err != nil {
+		log.Fatalf("failed to create influx reporter: %v", err.Error())
+	}
+	defer influxReporter.FlushAndShutdown()
+
+	log.Print("Starting the load test.")
+	// Ramp up load from 1k to 3k in 2 minute steps.
+	const duration = 2 * time.Minute
+	url := fmt.Sprintf("http://load-test-%s.default.svc.cluster.local?sleep=100", *flavor)
+	targeter := vegeta.NewStaticTargeter(vegeta.Target{
+		Method: http.MethodGet,
+		URL:    url,
+	})
+
+	// Make sure the target is ready before sending the large amount of requests.
+	if err := performance.ProbeTargetTillReady(url, duration); err != nil {
+		log.Fatalf("Failed to get target ready for attacking: %v", err)
+	}
+	// Wait for scale back to 0
+	if err := performance.WaitForScaleToZero(ctx, namespace, selector, 2*time.Minute); err != nil {
+		log.Fatalf("Failed to wait for scale-to-0: %v", err)
+	}
+
+	// Run vegeta attack with custom pacers and process results from channel
+	pacers := make([]vegeta.Pacer, 3)
+	durations := make([]time.Duration, 3)
+	for i := 1; i < 4; i++ {
+		pacers[i-1] = vegeta.Rate{Freq: i, Per: time.Millisecond}
+		durations[i-1] = duration
+	}
+	pacer, err := pkgpacers.NewCombined(pacers, durations)
+	if err != nil {
+		log.Fatalf("Error creating the pacer: %v", err)
+	}
+	resultsChan := vegeta.NewAttacker().Attack(targeter, pacer, 3*duration, "load-test")
+	metricResults := processResults(ctx, resultsChan, influxReporter, selector)
+
+	// Report the results
+	influxReporter.AddDataPointsForMetrics(metricResults, benchmarkName)
+	_ = vegeta.NewTextReporter(metricResults).Report(os.Stdout)
+
+	if err := checkSLA(metricResults); err != nil {
+		// make sure to still write the stats
+		influxReporter.FlushAndShutdown()
+		log.Fatalf(err.Error())
+	}
+
+	log.Println("Load test finished")
+}
+
+func processResults(ctx context.Context, results <-chan *vegeta.Result, reporter *performance.InfluxReporter, selector labels.Selector) *vegeta.Metrics {
+	ctx, cancel := context.WithCancel(ctx)
+	deploymentStatus := performance.FetchDeploymentsStatus(ctx, namespace, selector, time.Second)
+	sksMode := performance.FetchSKSStatus(ctx, namespace, selector, time.Second)
+	defer cancel()
+
+	metricResults := &vegeta.Metrics{}
+
+	for {
+		select {
+		case <-ctx.Done():
+			// If we time out or the pod gets shutdown via SIGTERM then start to
+			// clean thing up.
+			return nil
+
+		case res, ok := <-results:
+			if ok {
+				metricResults.Add(res)
+			} else {
+				// If there are no more results, then we're done!
+				// Compute latency percentiles
+				metricResults.Close()
+
+				return metricResults
+			}
+
+		case ds := <-deploymentStatus:
+			// Add a sample point for the deployment status
+			reporter.AddDataPoint(benchmarkName,
+				map[string]interface{}{"ready-replicas": float64(ds.ReadyReplicas), "desired-replicas": float64(ds.DesiredReplicas)})
```

Comment:
desired replicas and ready replicas are int32 I guess we are using float64 in order to have one function that adds a point to InfluxDB.

Reply:
No idea. I think it was like this before, but yes we probably can just report int32 to influx?